### ISSUE TITLE
IgxGrid isLoading needs data reset to work properly

### ISSUE THREAD

**Issue by bmtheo** [IgniteUI/igniteui-angular#5517]
## Description  
IgxGrid isLoading needs a data reset to work properly (see full detail on official forum : https://www.infragistics.com/community/forums/f/ignite-ui-for-angular/120300/igxgrid-isloading-is-working-only-the-first-time)

 * igniteui-angular version: 8.0.2
 * browser: Chrome/Edge

Full version
  "dependencies": {
    "@angular/animations": "^8.0.3",
    "@angular/cdk": "^8.0.1",
    "@angular/common": "^8.0.3",
    "@angular/compiler": "^8.0.3",
    "@angular/core": "^8.0.3",
    "@angular/forms": "^8.0.3",
    "@angular/router": "^8.0.3",
    "@ngrx/effects": "^8.0.1",
    "@ngrx/store": "^8.0.1",
    "igniteui-angular": "^8.0.2",
    "rxjs": "~6.5.2",
...
  },

## Steps to reproduce  

IgxGrid code : 
```
<igx-grid #f6Grid [data]="opaleForms$ | async" [isLoading]="isLoading$ | async>
<igx-column [resizable]="true" [field]="'Code'"></igx-column>
Some other columns...
</igx-grid>
```

User actions : 

1. View is loaded and everything is displayed with empty values (ie : The grid is here with the column but not values in it)
2. The user click on search button
3. The isLoading is set to true
4. The grid is displaying the spinner (loading works :+1:  )
5. The data is returned, the isLoading is set to false
6. The grid is displaying data and not the spinner (still good :+1: )
7. The user click again on the search button to search something else
8. The isLoading is set to true
9. the grid is not displaying the spinner (not working :cry: )

## Result  
The loading is not active (spinner is not displayed)

## Expected result  
The loading should be active

## Workaround
One workaround was found, at each time the isLoading is set to true, we set the data to an empty array like so : 

```
        this.isLoading$.pipe(takeUntil(this.unsubscribe)).subscribe(isLoading => {
            if (this.f6Grid) {
                if (isLoading) {
                    this.f6Grid.data = [];
                }
                this.f6Grid.isLoading = isLoading;
            }
        });
```

And this is working

To me this should not be necessary and the loading should work straight even if the data is set

**Comment by skrustev** [comment]
@bmtheo The loading indicator is intended to be used when the grid has no data and is waiting for it to be loaded, instead of appearing empty without any indication. You don't need to set `isLoading` to `false` after each search, since it currently acts as a substitute to the "Empty grid" template and will be hidden by default.
Clearing the data, like you've shown, is how this can be achieved as of now.

Your suggestion is valid though and we can look into improving it and enabling the loading indicator to be show even when there is data in the grid. This means though that it will be required for the developer to set `isLoading` to `false` after each time the data is loaded in order to hide it.

@radomirchev @kdinev @StefanIvanov 
I'll mark it as a feature request instead, since this will change the behavior of how the loading indicator should work.

**Comment by bmtheo** [comment]
@skrustev thanks for the information
Ok so I missunderstood the way isLoading work
It's indeed a feature request/enhancement

But I still think this is not a correct behavior and should be changed, the use cases are : 

- A new set of data is being fetched by the user (click on search button) -> We need to inform the user that the data is loading

- The data returned by the service is empty -> We need to inform the user that it's not loading anymore (error/message management are out of the grid scope I think)

In conclusion the loading state of the grid is not directly linked to the presence of data in igxgrid.data.
Do you agree on that statement ?

**Comment by skrustev** [comment]
@bmtheo Yes, in order for the grid to support this kind of scenarios for the loading indicator, the presence of the data shouldn't be taken into account.

**Comment by ChronosSF** [comment]
@StefanIvanov , @simeonoff ,

In order to achieve the customer requirements we'll need the loading indicator to be an overlay on top of the rendered rows. This requries new design (e.g. do we use a semi-transparent overlay over the whole grid / only the body, etc), colors, etc.

Once there is design I'll be able to very quickly get markup/typescript done that shows it while the `isLoading` flag is true.

I assume we'll still use the current template for when there is no data.

Also from an UX perspective - currently you don't need to touch isLoading once your data loads as the template is automatically replaced because it only shows while the data has 0 length. If it's a stand-alone flag as it should be for this functionality, should the grid try to set it to false when it sees the data length changing or we let the user handle it manually?

**Comment by StefanIvanov** [comment]
@ChronosSF do you think this issue is related to [#1578](https://github.com/IgniteUI/igniteui-angular/issues/1578)

**Comment by ChronosSF** [comment]
No, I don't think so. #1578 describes a remote virtualization scenario where loading happens on scroll. Hiding cell content in the way described there makes sense as the previous chunk of data is no longer valid for the new scroll position.

Here, I think, the customer expects to have the loading indicator that we currently show when there is no data:

![issue12](https://user-images.githubusercontent.com/2188411/63248648-7a919e00-c270-11e9-975c-b0ffded16136.gif)

As an overlay on top of loaded data if he manually enables it while making a call to his backend.

The current template is this:
```html
<ng-template #defaultLoadingGrid>
    <div class="igx-grid__loading">
        <igx-circular-bar [indeterminate]="true">
        </igx-circular-bar>
    </div>
</ng-template>
````

I'll try to scrape something together so that I can visually show you what I think the customer would expect.

**Comment by simeonoff** [comment]
@ChronosSF Yeah, we can create a container for it that will be hidden by default and show up only when triggered.

**Comment by StefanIvanov** [comment]
OK @ChronosSF I though we had a similar scenario that was triggered on search, rather than on scroll providing the chunk of data to display. Proceed as you and @simeonoff suggest.
